flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/expressioneditor/ExpressionEditorContext.java

```diff
@@ -301,19 +306,57 @@ public record Info(String expression, LinePosition startLine, int offset, JsonOb
                        String branch, String property) {
     }
 
-    public String fileUri() {
-        return fileUri;
-    }
+    /**
+     * Encapsulates document and import related context with lazy loading capabilities.
+     *
+     * @since 2.0.0
+     */
+    private static class DocumentContext {
 
-    public TextDocument textDocument() {
-        return document.textDocument();
-    }
+        private final WorkspaceManagerProxy workspaceManagerProxy;
+        private final String fileUri;
+        private final Path filePath;
 
-    public Path filePath() {
-        return filePath;
-    }
+        private Document document;
+        private List<ImportDeclarationNode> imports;
 
-    public WorkspaceManager workspaceManager() {
-        return workspaceManagerProxy.get(fileUri);
+        public DocumentContext(WorkspaceManagerProxy workspaceManagerProxy, String fileUri, Path filePath) {
+            this.workspaceManagerProxy = workspaceManagerProxy;
+            this.fileUri = fileUri;
+            this.filePath = filePath;
+        }
```

Comment:
Can simplify as follows
```suggestion
        public DocumentContext(WorkspaceManagerProxy workspaceManagerProxy, String fileUri, Path filePath) {
             this(workspaceManagerProxy, fileUri, filePath, null);
        }
```